src/tests/ftest/osa/osa_offline_parallel_test.py

```diff
@@ -0,0 +1,205 @@
+try:
+    # python 3.x
+    import queue as queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+class OSAOfflineParallelTest(TestWithServers):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: This test runs
+    daos_server offline drain,reintegration,
+    extend test cases in parallel.
+
+    :avocado: recursive
+    """
+    def setUp(self):
+        """Set up for test case."""
+        super(OSAOfflineParallelTest, self).setUp()
+        self.dmg_command = self.get_dmg_command()
+        self.no_of_dkeys = self.params.get("no_of_dkeys", '/run/dkeys/*')[0]
+        self.no_of_akeys = self.params.get("no_of_akeys", '/run/akeys/*')[0]
+        self.record_length = self.params.get("length", '/run/record/*')[0]
+        self.out_queue = queue.Queue()
+
+    @fail_on(CommandFailure)
+    def get_pool_leader(self):
+        """Get the pool leader
+           Returns :
+            int: pool leader value
+        """
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][3])
+
+    @fail_on(CommandFailure)
+    def get_pool_version(self):
+        """Get the pool version
+           Returns :
+            int : pool_version_value
+        """
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][4])
+
+    @fail_on(DaosApiError)
+    def write_single_object(self):
+        """Write some data to the existing pool."""
+        self.pool.connect(2)
+        csum = self.params.get("enable_checksum", '/run/container/*')
+        container = DaosContainer(self.context)
+        input_param = container.cont_input_values
+        input_param.enable_chksum = csum
+        container.create(poh=self.pool.pool.handle,
+                         con_prop=input_param)
+        container.open()
+        obj = DaosObj(self.context, container)
+        obj.create(objcls=1)
+        obj.open()
+        ioreq = IORequest(self.context,
+                          container,
+                          obj, objtype=4)
+        self.log.info("Writing the Single Dataset")
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length)
+                d_key_value = "dkey {0}".format(dkey)
+                c_dkey = ctypes.create_string_buffer(d_key_value)
+                a_key_value = "akey {0}".format(akey)
+                c_akey = ctypes.create_string_buffer(a_key_value)
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+
+    def dmg_thread(self, puuid, rank, target, action, results):
+        self.log.info("Action: {0}".format(action))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments